test/e2e/list_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"github.com/stretchr/testify/require"
+	"strings"
+	"testing"
+)
+
+func TestAppListing(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+	yaml1 := `
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: redis-primary
+  labels:
+    x: "y"
+spec:
+  ports:
+  - port: 6380
+    targetPort: 6380
+  selector:
+    app: redis
+    tier: backend
+`
+	yaml2 := `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: redis-config
+  labels:
+    x: "z"
+data:
+  key: value
+`
+
+	name := "test-app-list"
+	name2 := "test-app-list-2"
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+		kapp.Run([]string{"delete", "-a", name2})
+	}
+
+	cleanUp()
+	defer cleanUp()
+	logger.Section("App listing and filter label", func() {
+		out, _ := kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name,
+			"--labels", "x=y"}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
+		expectedOutput1 := `
```

Comment:
We should be checking JSON output here

Reply:
I think it's not necessary to test this output as we are just adding the label to the app here and therefore it's just a generic deployment summary which is tested in many places.

Reply:
removed.